pkg/build/controller/config_controller.go

```diff
@@ -79,7 +79,7 @@ func (c *BuildConfigController) HandleBuildConfig(bc *buildapi.BuildConfig) erro
 		if kerrors.IsConflict(err) {
 			instantiateErr = fmt.Errorf("unable to instantiate Build for BuildConfig %s/%s due to a conflicting update: %v", bc.Namespace, bc.Name, err)
 			utilruntime.HandleError(instantiateErr)
-		} else if buildgenerator.IsFatal(err) {
+		} else if buildgenerator.IsFatal(err) || kerrors.IsNotFound(err) || kerrors.IsBadRequest(err) {
```

Comment:
I'm just asking if you're confident we're not doing it wrong.

On Wed, Jul 13, 2016 at 5:31 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/build/controller/config_controller.go
> <https://github.com/openshift/origin/pull/9828#discussion_r70712740>:
>
> > @@ -79,7 +79,7 @@ func (c *BuildConfigController) HandleBuildConfig(bc *buildapi.BuildConfig) erro
> >  		if kerrors.IsConflict(err) {
> >  			instantiateErr = fmt.Errorf("unable to instantiate Build for BuildConfig %s/%s due to a conflicting update: %v", bc.Namespace, bc.Name, err)
> >  			utilruntime.HandleError(instantiateErr)
> > -		} else if buildgenerator.IsFatal(err) {
> > +		} else if buildgenerator.IsFatal(err) || kerrors.IsNotFound(err) || kerrors.IsBadRequest(err) {
>
> those methods don't return a badrequest, but if they did i would assume
> so... if a request is bad, is there a reason to think retrying it will help?
>
> if we're returning "badrequest" on something that isn't actually a bad
> request, then we're doing it wrong, i think.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9828/files/64a7ffb0efa0dc6bf954fe8e43d249281a80c7ab#r70712740>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p0elqKqKUHULUd0Fey3_d5W2cbi9ks5qVVkYgaJpZM4JL2_i>
> .
>